probe-rs/src/architecture/arm/ap/memory_ap/amba_ahb3.rs

```diff
@@ -0,0 +1,191 @@
+use crate::architecture::arm::{
+    ap::{AccessPortType, ApAccess, ApRegAccess},
+    communication_interface::RegisterParseError,
+    ArmError, DapAccess, FullyQualifiedApAddress, Register,
+};
+
+use super::{AddressIncrement, DataSize};
+
+/// Memory AP
+///
+/// The memory AP can be used to access a memory-mapped
+/// set of debug resources of the attached system.
+#[derive(Debug)]
+pub struct AmbaAhb3 {
+    address: FullyQualifiedApAddress,
+    csw: CSW,
+    cfg: super::registers::CFG,
+}
+
+impl AmbaAhb3 {
+    /// Creates a new AmbaAhb3 with `address` as base address.
+    pub fn new<P: DapAccess>(
+        probe: &mut P,
+        address: FullyQualifiedApAddress,
+    ) -> Result<Self, ArmError> {
+        use crate::architecture::arm::Register;
+        let csw = probe.read_raw_ap_register(&address, CSW::ADDRESS)?;
+        let cfg = probe.read_raw_ap_register(&address, super::registers::CFG::ADDRESS)?;
+        let (csw, cfg) = (csw.try_into()?, cfg.try_into()?);
+
+        let me = Self { address, csw, cfg };
+        let csw = CSW {
+            AddrInc: AddressIncrement::Single,
+            ..me.csw
+        };
+        probe.write_ap_register(&me, csw)?;
+        Ok(Self { csw, ..me })
+    }
+}
+
+impl super::MemoryApType for AmbaAhb3 {
+    type CSW = CSW;
+
+    fn status<P: ApAccess + ?Sized>(&mut self, probe: &mut P) -> Result<CSW, ArmError> {
+        static_assertions::const_assert_eq!(super::registers::CSW::ADDRESS, CSW::ADDRESS);
```

Comment:
How about we try and use `const { assert!(...) }` here, now that inline const is stable? :) We don't really need a new dependency for this. Note that assert_eq won't work.

Reply:
Okay clippy, what the heck